cinema/urls.py

```diff
@@ -1,10 +1,46 @@
-from django.urls import path
+from django.urls import path, include
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    MovieViewSet,
+    movie_list,
+    movie_detail,
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail,
+    CinemaHallViewSet
+)
+from rest_framework import routers
+
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet, basename="movie")
+
+cinema_hall_list = CinemaHallViewSet.as_view(actions={
+    "get": "list", "post": "create"
+})
+cinema_hall_detail = CinemaHallViewSet.as_view({
+    "get": "retrieve",
+    "put": "update",
+    "patch": "partial_update",
+    "delete": "destroy",
+})
 
 urlpatterns = [
     path("movies/", movie_list, name="movie-list"),
     path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("", include(router.urls)),
```

Comment:
No need to include router if base path is not provided just do:
urlpatterns = [list of urls] + router.urls